pkg/services/rendering/rendering.go

```diff
@@ -190,6 +188,43 @@ func (rs *RenderingService) render(ctx context.Context, opts Opts) (*RenderResul
 	return rs.renderAction(ctx, renderKey, opts)
 }
 
+func (rs *RenderingService) RenderCSV(ctx context.Context, opts CSVOpts) (*RenderCSVResult, error) {
+	startTime := time.Now()
+	result, err := rs.renderCSV(ctx, opts)
+
+	elapsedTime := time.Since(startTime).Milliseconds()
+	saveMetrics(elapsedTime, err)
+
+	return result, err
+}
+
+func (rs *RenderingService) renderCSV(ctx context.Context, opts CSVOpts) (*RenderCSVResult, error) {
+	if rs.inProgressCount > opts.ConcurrentLimit {
```

Comment:
I usually get confused by speaking about "concurrent" whatever, because when you only have a single one, there's no concurrency at all. But, I know that sometimes, the amount of "concurrent" whatever is like the max amount of whatever. For example, when speaking about "concurrent sessions", if that limit is set to 1, then you never could have two sessions coexisting.

That being said, if I understood this conditional statement properly, I'd say we could end up with `opts.ConcurrentLimit+1` in progress tasks. So, the question would be: is that what we actually want? 🤔 

Thanks!! 🙏🏻 

PS: Sorry for the long message, but I have some kind of dyslexia with this kind of concepts 🙈

Reply:
The description from the doc is: `# Set maximum number of concurrent calls to the rendering service`. I understand your confusion but, when I'm reading this and set it to 4, I expect that maximum 4 calls can be made at the same time to the rendering service. So I'd say we should change this to `if rs.inProgressCount >= opts.ConcurrentLimit {`

Reply:
Maybe I should make this change in another PR so this appears clearly in the changelog, WDYT @joanlopez @marefr @aknuds1?